gensim/similarities/docsim.py

```diff
@@ -190,7 +190,11 @@ def __init__(self, output_prefix, corpus, num_features, num_best=None, chunksize
             # undocumented feature: set output_prefix=None to create the server in temp
             self.output_prefix = utils.randfname(prefix='simserver')
         else:
-            self.output_prefix = output_prefix
+            self.output_prefix = os.path.join(os.path.dirname(output_prefix), 'shard\\' if os.name == 'nt' else 'shard/')
```

Comment:
``` os.path.join ( os.path.dirname('x/y/z/abc'),'shard', '') ``` will give 
```x/y/z/shard/ ```

we want everything to happen inside shard directory. For example, we want shard index to be saved as ```/x/y/z/shard/.0``` not as ```x/y/z/shard.0``` which is the case in absence of leading slash.

Earlier, if user had passed path as self.output_index to be ```x/y/z/abc``` the shard would be ```x/y/z/abc.0``` and later user may provide some other path for index, so the idea is to have both in same directory.